auger_cli/api/experiments.py

```diff
@@ -0,0 +1,247 @@
+        experiments_list = client.call_hub_api(['experiments', 'list'],
+            params={'name': experiment_name, 'project_id': project_id, 'limit': REQUEST_LIMIT}
+        )
+        if len(experiments_list) > 0:
+            result = experiments_list[0]
+
+    return result
+
+
+def create(client, name, project_id, data_path):
+    return client.call_hub_api(['experiments', 'create'],
+        params={
+            'id': "experiment_" + name + "_" + get_uid(),
+            'name': name,
+            'project_id': project_id,
+            'data_path': data_path
+        }
+    )
+
+
+def delete(client, experiment_id):
+    experiment = client.call_hub_api(['experiments', 'delete'],
+        params={'id': experiment_id}
+    )
+
+def update(client, experiment_id, name):
+    return client.call_hub_api(['experiments', 'update'],
+        params={
+            'id': experiment_id,
+            'name': name
+        }
+    )
+
+
+def get_or_create(client, project_id):
+    experiment_id, experiment_name = client.config.get_experiment()
+
+    if experiment_id is not None:
+        return read(client, experiment_id=experiment_id)
+
+    experiment = read(client, project_id, experiment_name)
+    if experiment.get('id') is not None:
+        return experiment
+
+    if len(client.config.get_evaluation_options()) == 0:
+        return {}
+
+    if len(client.config.get_evaluation_options().get('data_path', ''))==0:
+        client.print_line("To create experiment {}, evaluation_options should contain data_path.".format(experiment_name))
+        return {}
+
+    client.print_line(
+        'Experiment {} does not exist. Creating ...'.format(experiment_name))
+
+    # search_space = create_cluster_task_ex(client, project_id,
+    #     "auger_ml.tasks_queue.tasks.get_experiment_configs_task",
+    #     {'augerInfo':{'experiment_id': None}}
+    # )
+    return create(
+        client, experiment_name, project_id, client.config.get_evaluation_options()['data_path'])
+
+
+def read_ex(client, experiment_id=None):
+    if experiment_id is not None:
+        experiment = read(client, experiment_id=experiment_id)
+        project = projects.read(client, project_id=experiment.get('project_id'))
+    else:    
+        project = projects.get_or_create(client, create_if_not_exist=True)
+        experiment = get_or_create(client, project['id'])
+
+    result = experiment    
+    if project.get('cluster_id'):
+        result['cluster'] = clusters.read(client, project.get('cluster_id'), clusters.display_attributes)
+
+    if client.config.get_experiment_session_id():    
+        result['session'] = experiment_sessions.read(client, client.config.get_experiment_session_id(), ['id', 'status', 'datasource_name', 'model_type'])
+
+    return result
+
+def run(client):
+    client.config.delete_session_file()
+
+    project_id = projects.start(client, create_if_not_exist=True)
+
+    experiment = get_or_create(client, project_id)
+
```

Comment:
I created some example rest calls but look at api reference as well.

https://app.auger.ai/api/v1/docs
https://github.com/deeplearninc/auger-cli/pull/33/files#diff-c1d681950d6cde9754aae1947c2d55feR118

You should create experiment session here and then use task_args as model_settings argument for experiment_session.

Reply:
Who is going to create dataset manifest? And more important who is tracking changes between runs? Now I do it in this task:
is_dataset_changed_task

To create exp_session we need old or new dataset manifest

Reply:
I use trials APi to get latest leaderboard, this is not an issue